Fix notification of field change in a PROTO not propagating through DEF/USE

ad-daniel opened the pull request:
**Description**
Fixes https://github.com/cyberbotics/webots/issues/4240

The bug doesn't affect just meshes, any shape with a primitive as geometry doesn't propagate either if within a PROTO.

The bug lies in [this](https://github.com/cyberbotics/webots/blob/f114270206dc12f345fb86c67adce3fa6a93ec84/src/webots/vrml/WbNode.cpp#L753-L799) function. When a field changes, be it manually or from supervisor, this function is supposed to propagate the change to all [USE-nodes](https://github.com/cyberbotics/webots/blob/f114270206dc12f345fb86c67adce3fa6a93ec84/src/webots/vrml/WbNode.cpp#L779-L781) as well.
The functions that determine whether this notification has to occur (findSubFieldIndex and findSubField) however only search for a matching field (i.e a field with same name and type as the one that triggered the notification) only at the top level. Namely, in this case it searches only in the fields of the `Shape` node, which for the base node case entails checking `Shape`, `PBRAppearance` and `Mesh` where ultimately it finds it and everything works as intended. If the `Shape` is however defined within a PROTO, only the PROTO itself is checked, without checking if the trigger might have been in the appearance or geometry fields itself (i.e the nodes `PBRAppearance` and `Mesh` are not checked), hence why the notification doesn't occur.

As far as I can tell, these functions are not used elsewhere so it should be safe to do directly the change withing them, by forcing the search also in the sub nodes themselves.

omichel reviewed (approved):
That looks good to me. Thank you.

ad-daniel commented:
I've also fixed the issue regarding the incorrect update of the mesh on the ODE side resulting from a change of url and added a sanity check for the url as well (a warning message is shown in case the resource cannot be found). This approach seemed the cleaner solution, it's pointless to even attempt to build a wren mesh (and everything that comes with it namely liberating the previous one from ODE, updating the mesh cache, ...) if we know the url is not valid to begin with.

It isn't necessary to add a test for this second aspect since it is implicitly being tested in https://github.com/cyberbotics/webots/pull/4238 (to test if the cache works this PR changes the mesh 3 times using a supervisor)

omichel reviewed (approved):
Well spotted!
Thank you.
